packages/dev/core/src/Collisions/pickingInfo.ts

```diff
@@ -105,9 +105,20 @@ export class PickingInfo {
             result = Vector3.Cross(p1p2, p3p2);
         }
 
-        // Flip the normal if the picking ray is in the same direction.
-        if (this.ray && Vector3.Dot(result, this.ray.direction) > 0) {
-            result.negateInPlace();
+        if (this.ray) {
+            const matMesh = TmpVectors.Matrix[0];
+            const rayDirInLocal = TmpVectors.Vector3[0];
+
+            // Transform the ray direction to mesh local space
+            matMesh.copyFrom(this.pickedMesh.getWorldMatrix());

```

Comment:
This sounds coastly. Is there no invert cache on the mesh ?

Reply:
Maybe we could move in world the result instead of going back to local so we prevent the invert need ?